drivers/include/mtd_default.h

```diff
@@ -25,31 +25,6 @@
 extern "C" {
 #endif
 
-#if defined(MODULE_MTD_SDCARD_DEFAULT)
```

Comment:
I would have seen `mtd_default.h` as a sink for including `mtd_sdcard.h`, `mtd_emulated.h`, `mtd_flashpage.h` ...

Reply:
I'd think so too, it's the generic MTD include that gets you whatever MTDs are provided by the board.

(We should also have a `mtd_default` pseudo-module similar to `netdev_default` - currently the `mtd` module also serves as that role)

Reply:
I wasn't sure if I should include it in this PR since this change is more general in nature and not related to the changes in this PR.

By the way, I had to notice that the changes of the PR in this file got a bit mixed up after some rebasing :wink:

Reply:
Ah now I understand, the `mtd_sdcard.h` etc includes are simply not necessary anymore - anything that accesses that will just use the generic mtd interface, the backend specific headers are pretty much internal.

So removing them from this file is fine.

Reply:
The following construct which was added by myself in PR #19538
```c
#if MTD_NUMOF > 0
extern mtd_dev_t *mtd0;
#endif
...
```
doesn't work any longer with this PR since `MTD_NUMOF` is now defined as the size of the array `mtd_dev_xfa`:
```
#define MTD_NUMOF   XFA_LEN(mtd_dev_t *, mtd_dev_xfa)
```
I'm wondering whether we should simply declare a certain number of `mtd<n>` variables independent on whether they are really defined by the backend. The only problem that can happen is that the application could try to use a `mtd<n>` that is not defined which then leads to a linking error.

Reply:
Compile time / link time errors are fine, as long as we don't add run-time errors I see no problem.

Reply:
I decided to complete missing `MTD_<n>` definitions for boards that have SD Cards and to use `MTD_<n>` to declare extern `mtd<n>` variables in `mtd_default.h`:
```c
#ifdef MTD_0
extern mtd_dev_t mtd0;
#endif
```

Reply:
@benpicco Since boards should define the `MTD_<n>` macros for their `mtd<what_ever>` variables, it seems to be more reasonable to declare the extern variables in `mtd_default.h` based on defined `MTD_<n>` macros instead of using fix `mtd<n>` declarations, see commmit 7f65c67e187c1ef50f109ed4cb1b85365394d1f3.

What do you think?

IMHO, we should also note in [MTD doc](https://doc.riot-os.org/group__drivers__mtd.html) that board definitions shall define `MTD_<n>` for their MTDs to have a consitent approach and to be able to use all features of the MTD API.